packages/platform-core/src/telemetry/usage_data_emitter.ts

```diff
@@ -0,0 +1,131 @@
+import { v4 as uuid } from 'uuid';
+import { AccountIdFetcher } from './account_id_fetcher.js';
+import { UsageData } from './usage_data.js';
+import os from 'os';
+import https from 'https';
+import { getInstallationUuid } from './get_installation_id.js';
+import { latestPayloadVersion } from './constants.js';
+import { getUrl } from './get_usage_data_url.js';
+import isCI from 'is-ci';
+import { SerializableError } from './serializable_error.js';
+
+/**
+ * Entry point for sending usage data metrics
+ */
+export class DefaultUsageDataEmitter {
```

Comment:
this should implement the type defined in the factory file

Reply:
updated